tests/functional/RuleTicket.php

```diff
@@ -1167,4 +1167,118 @@ public function testAssignLocationFromUser(
         $ticket->getFromDB($ticket->getID());
         $this->integer($ticket->fields['locations_id'])->isEqualTo($expected_location_after_creation);
     }
+
+    public function testAssignGroupOnUpdate()
```

Comment:
Test is not enough complete.

1. It should ensure that action like `replace users assigned` will have no impact on requesters and observers, nor on groups or suppliers.
2. It should verify that replacement also removes actors in existing input (not only existing actors).
3. It should verify that, when there is no existing actors, it does not fail.
4. It should verify the `ONADD` case too.

You should probably have a data provider that is used to test multiple combinations.